### DISCUSSION TITLE
Adding support for lockable resources

### DISCUSSION THREAD

**Discussion by urosctd** [connorads/lockbot#188] [Ideas]
This is a frequent case that keeps happening at my company.
People are using different names for the same resources (e.g. someone is adding `-service` at the end and someone isn't), also typos happen, we had a couple edge cases too.
What if we add support for adding lockable resources?

**Comment by connorads** [comment]
Hey thanks for the feedback/idea. Could you elaborate a bit on it. How would you expect it to work? Could you give some examples of the use case and what the commands and user responses look like

**Reply by urosceh** [reply]
Well maybe each channel would have smth like this:

/add-lockable-resource:
1. if the channel doesn't already have lockable resources, meaning it's not using them, you would need to pass some flag like init for example or even lbtoken, just to protect from someone mistakingly setting this
2. if the channel isn't using this, return just that the channel isn't using the locks

/lock:
1. when [fetching lockOwner](https://github.com/connorads/lockbot/blob/ecb4df985d9df65c92ead70d2f94ab58c1883eda/src/lock-bot.ts#L54C5-L54C77), get lockableResources
2. if the channel has lockable resources, should check if the resource is in lockable resources


I'm just checking if this idea resonates with you.
